ethers-solc/src/lib.rs

```diff
@@ -93,8 +93,58 @@ impl Project {
     ///
     /// NOTE: this does not check if the contracts were successfully compiled, see
     /// `CompilerOutput::has_error` instead.
+
+    /// NB: If the `svm` feature is enabled, this function will automatically detect
+    /// solc versions across files.
     pub fn compile(&self) -> Result<ProjectCompileOutput> {
-        let mut sources = self.sources()?;
+        let sources = self.sources()?;
+
+        #[cfg(not(all(feature = "svm", feature = "async")))]
+        {
+            self.compile_with_version(&self.solc, sources)
+        }
```

Comment:
should this be also configureable via some Project settings flag?

Reply:
I don't think so? Isn't it always better to support many?

Reply:
you're right, that's better!